lib/mayaUsd/commands/layerEditorCommand.cpp

```diff
@@ -901,24 +930,42 @@ class RefreshSystemLockLayer : public BaseCmd
             _refreshLayerSystemLock(layer);
         }
 
+        // Execute lock commands
```

Comment:
Do and undo stop at the first error and execute in the same order. For composite commands, the usual way to handle it is to record which were done so undo only undoes those that were actually successful. Also, undo usually execute in reverse order as do. Might be worthwhile to either execute as much as possible (execute all, return if any failed) or to undo on error.

For locking, these things are less critical, I don't think there will be anything bad so you can keep the simpler code that you currently have, but it is something to keep in mind.

Reply:
Great point!

In this implementation, the layer hierarchy is flattened and since locking only affects the local variables of each layer, the lock / unlock doesn't have to be in an order. I'll keep the error checking in mind going forward.